app/src/main/java/com/voipgrid/vialer/calling/NetworkAvailabilityActivity.java

```diff
@@ -0,0 +1,101 @@
+package com.voipgrid.vialer.calling;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.text.format.DateUtils;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.voipgrid.vialer.R;
+import com.voipgrid.vialer.VialerApplication;
+import com.voipgrid.vialer.sip.SipCall;
+import com.voipgrid.vialer.sip.SipService;
+import com.voipgrid.vialer.util.NetworkUtil;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import de.hdodenhof.circleimageview.CircleImageView;
+
+public class NetworkAvailabilityActivity extends AbstractCallActivity {
```

Comment:
Have you tested what happens when:

- You lose connectivity when the app is not in the foreground
- You leave the app after you have lost connectivity and then go back into it

Is what you expect/want to happen actually happening? I don't have the correct answer for this, probably for UX to decide but it's worth considering.